File: src/test/java/org/mpisws/concurrent/programs/inconsistent/counter/Counter.java
package org.mpisws.concurrent.programs.inconsistent.counter;

public class Counter {
    public int count;

    public Counter() {
        count = 0;
    }

    public int get() {
        return count;
    }

    public void set(int newValue) {
        count = newValue;
    }
}

File: src/main/java/org/example/concurrent/programs/kotlin/counter/Counter.kt
package org.example.concurrent.programs.kotlin.counter

import kotlin.concurrent.thread

class Counter {
    @Volatile
    var count = 0
}

fun main() {
    val counter = Counter()

    val thread1 = thread(start = true) {

        counter.count++

    }

    val thread2 = thread(start = true) {

        counter.count++

    }

    thread1.join()
    thread2.join()

    println("Final count: ${counter.count}")
}